ibm-cloud-shell/v1.ts

```diff
@@ -0,0 +1,342 @@
+   * @param {RegionSetting[]} [params.regions] - List of Cloud Shell region settings.
+   * @param {OutgoingHttpHeaders} [params.headers] - Custom request headers
+   * @returns {Promise<IbmCloudShellV1.Response<IbmCloudShellV1.AccountSettings>>}
+   */
+  public updateAccountSettings(
+    params: IbmCloudShellV1.UpdateAccountSettingsParams
+  ): Promise<IbmCloudShellV1.Response<IbmCloudShellV1.AccountSettings>> {
+    const _params = { ...params };
+    const requiredParams = ['accountId'];
+
+    const missingParams = getMissingParams(_params, requiredParams);
+    if (missingParams) {
+      return Promise.reject(missingParams);
+    }
+
+    const body = {
+      '_rev': _params.rev,
+      'default_enable_new_features': _params.defaultEnableNewFeatures,
+      'default_enable_new_regions': _params.defaultEnableNewRegions,
+      'enabled': _params.enabled,
+      'features': _params.features,
+      'regions': _params.regions,
+    };
+
+    const path = {
+      'account_id': _params.accountId,
+    };
+
+    const sdkHeaders = getSdkHeaders(
+      IbmCloudShellV1.DEFAULT_SERVICE_NAME,
+      'v1',
+      'updateAccountSettings'
+    );
+
+    const parameters = {
+      options: {
+        url: '/api/v1/user/accounts/{account_id}/settings',
+        method: 'POST',
+        body,
+        path,
+      },
+      defaultOptions: extend(true, {}, this.baseOptions, {
+        headers: extend(
+          true,
+          sdkHeaders,
+          {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+          },
+          _params.headers
+        ),
+      }),
+    };
+
+    return this.createRequest(parameters);
+  }
+}
+
+/*************************
+ * interfaces
+ ************************/
+
+namespace IbmCloudShellV1 {
+  /** An operation response. */
+  export interface Response<T = any> {
+    result: T;
+    status: number;
+    statusText: string;
+    headers: IncomingHttpHeaders;
+  }
+
+  /** The callback for a service request. */
+  export type Callback<T> = (error: any, response?: Response<T>) => void;
+
+  /** The body of a service request that returns no response data. */
+  export interface Empty {}
+
+  /** A standard JS object, defined to avoid the limitations of `Object` and `object` */
+  export interface JsonObject {
+    [key: string]: any;
+  }
+
+  /*************************
+   * request interfaces
+   ************************/
+
+  /** Parameters for the `getAccountSettings` operation. */
+  export interface GetAccountSettingsParams {
+    /** The account ID in which the account settings belong to. */
+    accountId: string;
+    headers?: OutgoingHttpHeaders;
+  }
+
+  /** Parameters for the `updateAccountSettings` operation. */
+  export interface UpdateAccountSettingsParams {
+    /** The account ID in which the account settings belong to. */
+    accountId: string;
+    /** Unique revision number for the settings object. */
+    rev?: string;
```

Comment:
These field names look much better after adding the readonly attributes to the AccountSettings "account_id" schema property. It avoids the naming conflict with the "account_id" path param and so no prefixing the names of body-related properties with "new_".  :+1: